src/main/java/org/jboss/tools/intellij/componentanalysis/CAIntentionAction.java

```diff
@@ -52,7 +57,19 @@ protected CAIntentionAction(PsiElement element, VulnerabilitySource source, Depe
 
     @Override
     public void invoke(@NotNull Project project, Editor editor, PsiFile file) throws IncorrectOperationException {
-        this.updateVersion(project, editor, file, getRecommendedVersion(this.report));
+        String recommendedVersion = getRecommendedVersion(this.report);
+        this.updateVersion(project, editor, file, recommendedVersion);
+        ApiService apiService = ServiceManager.getService(ApiService.class);
+        SaUtils saUtils = new SaUtils();
+        var telemetryMsg = TelemetryService.instance().action("component_analysis_recommendation_accepted");
+        telemetryMsg.property("package", this.report.getRef().name());
+        telemetryMsg.property("version", recommendedVersion);
+        telemetryMsg.property(ApiService.TelemetryKeys.ECOSYSTEM.toString(), saUtils.determinePackageManagerName(file.getName()));
```

Comment:
```suggestion
        telemetryMsg.property(ApiService.TelemetryKeys.ECOSYSTEM.toString(), CAAnnotator.getPackageManager(file.getName()));
```
Please use this [method](https://github.com/redhat-developer/intellij-dependency-analytics/blob/1b84290776669494558b80a2ccedd82258b6e6d1/src/main/java/org/jboss/tools/intellij/componentanalysis/CAAnnotator.java#L244) instead and change it to be `static`. Also delete line 63.

Reply:
Done